lib/palindrome_check.rb

```diff
@@ -1,7 +1,40 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the lenght of my_phrase
+# Space complexity: O(1) because no additional space is needed
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  return false if my_phrase.nil?
+  i = 0
+  j = my_phrase.length - 1
+
+  while i < j
+    if my_phrase[i] == my_phrase[j]
+      i += 1
+      j -= 1
+    elsif my_phrase[i] == " "
+      i += 1
+    elsif my_phrase[j] == " "
+      j -= 1
+    elsif my_phrase[i] != my_phrase[j] && my_phrase[i] != " " && my_phrase[j] != " "
```

Comment:
will `my_phrase[i] != " " && my_phrase[j] != " "` ever be true considering you are guarding against spaces on lines 14 and 16?